### ISSUE TITLE
Add optional fields support

### ISSUE THREAD

**Issue by vgarvardt** [grpc-ecosystem/protoc-gen-grpc-gateway-ts#21]
As of [protobuf 3.15](https://github.com/protocolbuffers/protobuf/releases/tag/v3.15.0) `optional` fields are available, but they are not supported by the generator:

```
test/optional/v1/has_optional_field.proto: is a proto3 file that contains optional fields, but code generator protoc-gen-grpc-gateway-ts hasn't been updated to support optional fields in proto3. Please ask the owner of this code generator to support proto3 optional.--grpc-gateway-ts_out:
```

`test/optional/v1/has_optional_field.proto` contents:

```proto
syntax = "proto3";

package test.optional.v1;

message HasOptionalField {
  string not_optional = 1;
  optional string not_required = 2;
}
```

**Comment by lyonlai** [comment]
haven't run into this before. let me have a look and get back to you later.

**Comment by lyonlai** [comment]
👋  @vgarvardt getting around the error itself is simple. But that doesn't mean the output code will have full support of the optional semantic in proto3. So I don't think it's the right thing to do.

The current situation is that the support of proto2 optional/require is not properly implemented in this generator. To achieve the real support of optional field in proto3, might as well solve the proto2 problem at the same time. So there will be quite a bit of work needs to be done.

**Comment by vgarvardt** [comment]
I'm still very interested in this and may spend some time helping with the implementation, but I'll need your guidance. Maybe you can describe the job that needs to be done for a proper implementation, or at least the first steps so I could try to work on it and let you review PRs if you do not have to work on it.

**Comment by yinheli** [comment]
@vgarvardt  I just let `optional ` enable to generate, but not implements, refer: https://github.com/yinheli/protoc-gen-grpc-gateway-ts/commit/bb766c2d072ebc800aaac902add2c330a27bca72

**Comment by lyonlai** [comment]
yup, what @yinheli suggested will  unblock you for now. @vgarvardt. I was gonna start working on it but got distracted to other priorities. Free free to start thinking about it before the implementation. I'll guide you through

**Comment by ergoz** [comment]
any updates? It will make usable this generator. It is really sucks to use grpc-web instead....

**Comment by theartofdevel** [comment]
any updates?

**Comment by KubeKyrie** [comment]
any updates?
